contracts/strategies/donation-voting/DonationVotingStrategy.sol

```diff
@@ -0,0 +1,451 @@
+            _allocationStartTime,
+            _allocationEndTime,
+            _allowedTokens
+        );
+    }
+
+    function __DonationVotingStrategy_init(
+        uint256 _poolId,
+        bool _registryGating,
+        uint256 _registrationStartTime,
+        uint256 _registrationEndTime,
+        uint256 _allocationStartTime,
+        uint256 _allocationEndTime,
+        address[] memory _allowedTokens
+    ) internal {
+        __BaseStrategy_init(_poolId);
+        registryGating = _registryGating;
+        registrationStartTime = _registrationStartTime;
+        registrationEndTime = _registrationEndTime;
+        allocationStartTime = _allocationStartTime;
+        allocationEndTime = _allocationEndTime;
+
+        uint256 allowedTokensLength = allowedTokens.length;
+        for (uint256 i = 0; i < allowedTokensLength; i++) {
+            allowedTokens.push(_allowedTokens[i]);
+        }
+    }
+
+    /// ===============================
+    /// ============ Views ============
+    /// ===============================
+
+    /// @notice Get the recipient
+    /// @param _recipientId Id of the recipient
+    function getRecipient(address _recipientId) external view returns (Recipient memory) {
+        return _getRecipient(_recipientId);
+    }
+
+    /// @notice Get Internal recipient status
+    /// @param _recipientId Id of the recipient
+    function getInternalRecipientStatus(address _recipientId) external view returns (InternalRecipientStatus) {
+        return _getRecipient(_recipientId).recipientStatus;
+    }
+
+    /// @notice Get recipient status
+    /// @param _recipientId Id of the recipient
+    function getRecipientStatus(address _recipientId) external view override returns (RecipientStatus) {
+        InternalRecipientStatus internalStatus = _getRecipient(_recipientId).recipientStatus;
+        if (internalStatus == InternalRecipientStatus.Appealed) {
+            return RecipientStatus.Pending;
+        } else {
+            return RecipientStatus(uint8(internalStatus));
+        }
+    }
+
+    /// @notice Returns the payout summary for the accepted recipient
+    function getPayouts(address[] memory __recipientIds, bytes memory, address)
+        external
+        view
+        returns (PayoutSummary[] memory payouts)
+    {
+        uint256 recipientLength = __recipientIds.length;
+
+        payouts = new PayoutSummary[](recipientLength);
+
+        for (uint256 i = 0; i < recipientLength;) {
+            payouts[i] = payoutSummaries[__recipientIds[i]];
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Checks if address is elgible allocator
+    function isValidAllocator(address) external pure returns (bool) {
+        return true;
+    }
+
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Review recipient application
+    /// @param __recipientIds Ids of the recipients
```

Comment:
some parameters in the whole code have two underscores instead of one.

Reply:
that's cause _recipientIds is global private variable. Got a nice way to handle this ? Maybe a better name ?

Reply:
make `_recipientIds` public :laughing: 

we could also introduce `recipientIds_` for global variables

Reply:
okie this works recipientIds_